py/vm.c

```diff
@@ -1355,6 +1355,16 @@ unwind_jump:;
                 }
             }
 
+            #if MICROPY_PY_SYS_UATEXIT
```

Comment:
I don't think it's right to put this code here.  atexit functions should probably be handled on a per-port basis, because the port defines what is meant by "exit" (it would be soft reset on most bare-metal ports).

Reply:
accepted